app/models/user.rb

```diff
@@ -0,0 +1,57 @@
+class User < ActiveRecord::Base
+  devise :registerable, :confirmable,  :database_authenticatable,
+         :recoverable,  :rememberable, :validatable
+  
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  
+  has_many :comments,       dependent:    :destroy
+  has_many :microposts,     dependent:    :destroy
+  has_many :notifications,  dependent:    :destroy
+  
+  has_many :notification_senders,   class_name:  "Notification",
+                                    foreign_key: "sender_id",
+                                    dependent:   :destroy
+  has_many :active_relationships,   class_name:  "Relationship",
+                                    foreign_key: "follower_id",
+                                    dependent:   :destroy
+  has_many :passive_relationships,  class_name:  "Relationship",
+                                    foreign_key: "followed_id",
+                                    dependent:   :destroy
+                                    
+  has_many :following, through: :active_relationships,  source: :followed
+  has_many :followers, through: :passive_relationships, source: :follower
+  
+  before_save { self.email = email.downcase }
+  
+  validates :name,  presence: true, length: { maximum: 50 }
+  validates :email, presence: true, length: { maximum: 255 },
```

Comment:
can remove email regex and validations for email/password because devise already does that.